src/zarr/buffer/__init__.py

```diff
@@ -0,0 +1,12 @@
+"""
+Public API for implementations of the Zarr Buffer interface.
+
+See Also
+========
+arr.abc.buffer: Abstract base class for the Zarr Buffer interface.
+"""
+
+from ..core.buffer import default_buffer_prototype
+from . import cpu, gpu
```

Comment:
I don't think we import submodules like this anywhere else? If we don't, to be consistent I would remove these imports, and we can always discuss doing this throughout the package in an issue.

Reply:
It doesn't look like `cpu` and `gpu` have been removed as of the latest commit? Because we don't do this kind of import elsewhere (e.g., see `zarr/api/__init__.py`, that doesn't import `synchronous` or `asyncrhonous`), I think we should be consistent and not import `cpu` or `gpu` here.

Reply:
Sorry I misunderstood your earlier comment.

This is to ease migration for users: Previously `import zarr` would make, e.g. `zarr.core.buffer.cpu.Buffer` available, as a side-effect of us importing modules internally.

The closest analog here is `zarr.storage`, which imports all store subclasses, rather than making users use the full import path. We can't do that exact same thing here hough, since both `.gpu` and `.cpu` use the same names (Buffer, NDBuffer, etc), so the imports need to be namespaced.
 
We could just not import `cpu` and `gpu` and make the user do that import, but in this case I think consistency with `zarr.store` is most important.